.storybook/decorators/PropCombinations/README.md

```diff
@@ -0,0 +1,275 @@
+# Prop Combinations Decorator
+
+## Motivation
+
+Storybook & Chromatic provide no way to programmatically define stories for multiple variants. For example, to test the `Button` component with every combination of variant, size, darkMode, icon position etc., we would have to explicitly write each of these stories in the `*.stories.tsx file`, resulting in a lot of manual overhead work. In addition, testing each of these individually would result in hundreds, or even thousands, of stories which would dramatically increase the cost of Chromatic, and generates noise on the sidebar.
+
+As an alternative, this custom decorator reads metadata from the storybook default export, and generates a single story that renders all combinations of props defined in the story metadata.
+
+# Getting started
+
+## Storybook Config
+
+Add a `generate` parameter in the story meta object.
+
+```tsx
+// Button.stories.tsx
+const meta: StoryMeta<typeof Button> = {
+  component: Button,
+  parameters: {
+    default: 'LiveExample',
+    // Define combinations of props to generate
+    generate: {
+      props: {
+        darkMode: [false, true],
+        rightGlyph: [undefined, <Icon glyph={'ArrowRight'} />],
+        leftGlyph: [undefined, <Icon glyph={'Cloud'} />],
+        children: ['MongoDB', undefined],
+        size: Object.values(Size),
+        variant: Object.values(Variant),
+      },
+    },
+  },
+};
+export default meta;
+```
+
+Export an empty story titled `Generated` from the story file
+
+```tsx
+// Button.stories.tsx
+...
+export const Generated = () => {};
+```
+
+This will create a story called "Generated", that renders all defined prop combinations.
+
+![Decorator Demo](./docs/decorator-demo.png)
+
+## Multiple generated stories
+
+You can also break up the generated prop combinations into multiple stories.
+
+To do this, list the generated story names in the `generate` parameter, and set the args you want to remain static on the individual story.
+
+```tsx
+const meta: StoryMeta<typeof Button> = {
+  ...
+  parameters: {
+    generate: {
+      stories: ['LargeSize', 'DefaultSize'],
+      props: { ... },
+    },
+  }
+}
+
+...
+
+export const DefaultSize: StoryType<typeof Button> = () => <></>;
+DefaultSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Default,
+    },
+  },
+};
+
+export const LargeSize: StoryType<typeof Button> = () => <></>;
+LargeSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Large,
+    },
+  },
+};
+```
+
+# Options
+
+## `args`
+
+`Record<key in keyof ComponentProps, ComponentProps[key]>`
```

Comment:
Imo showing TS types within documentation isn't that helpful without IntelliSense. Can we include this in a more human-readable format?

Reply:
"Object of prop keys"